File: cpp/0541反转字符串Ⅱ.cpp
#include<string>
#include<iostream>
#include<algorithm>
using namespace std;

string reverseStr(string s, int k)
{

    for (int i = 0;i<s.size();i+=2*k)
    {
        if(i+k>s.size())
        {
            reverse(s.begin() + i, s.end());
            break;
        }
        reverse(s.begin() + i, s.begin() + i + k);
    }

    return s;
}

int main()
{
    string s = "123";
    reverse(s.begin(), s.begin() + 4);
    cout << s;
}

File: python/0541.反转字符串Ⅱ.py


def reverseStr(s: str, k: int) -> str:
    res=""
    
    while(True):  
        if(not len(s)):
            break
        
        res+=(s[k-1::-1]+s[k:2*k])
        s=s[2*k:]

    return res

s="0123456789"
# print(s[2::-1])
print(reverseStr(s,2))